### PR TITLE
`solve` now acknowledges `dict=True` for Relational args

### LINKED ISSUE #13534: solve dict=True ignored
```
In [22]: solve(Eq(2*x,1), dict=true)
Out[22]: [{x: 1/2}]

In [23]: solve([Eq(x**2-1,0),Gt(x,0)],(x,),dict=True)
Out[23]: x = 1
```

### PR DESCRIPTION
**Pull request by czgdp1807**
#### References to other Issues or PRs

Fixes #13534
Closes #13540

#### Release Notes

* solvers
  * `solve` now acknowledges `dict=True` for Relational args

### CODE REVIEW

**Comment by czgdp1807** [comment]
ping @jksuom @asmeurer

**Comment by czgdp1807** [comment]
What should `solve([Eq(x, x), Eq(x, x+1)], x, dict=True)` return? `[]` or `[{}]`?

File: sympy/solvers/tests/test_solvers.py [hunk 1]
```diff
@@ -167,14 +167,24 @@ def test_solve_args():
     assert solve((x + y - 2, 2*x + 2*y - 4)) == {x: -y + 2}
     # When one or more args are Boolean
     assert solve([True, Eq(x, 0)], [x], dict=True) == [{x: 0}]
-    assert solve([Eq(x, x), Eq(x, 0), Eq(x, x+1)], [x], dict=True) == []
+    assert solve([Eq(x, x), Eq(x, 0), Eq(x, x+1)], [x], dict=True) == [{}]
```
Comment (oscarbenjamin) [inline sympy/solvers/tests/test_solvers.py]: The previous output here was correct.

**Comment by czgdp1807** [comment]
Someone, please help me with, https://travis-ci.org/sympy/sympy/jobs/600244489 
I am unable to figure out why this is happening.

**Comment by czgdp1807** [comment]
> I am unable to figure out why this is happening.

Figured out. Thanks.

**Comment by czgdp1807** [comment]
Since, all the tests are passing, the PR will be merged after 24 hours, if no critical objections are raised within the mentioned period. Thanks.

File: sympy/solvers/tests/test_solvers.py [hunk 2]
```diff
@@ -175,6 +175,16 @@ def test_solve_args():
...
 
 
+def test_solve_dict():
+    assert solve(Eq(2*x,1), dict=True) == [{x: S(1)/2}]
+    assert solve([Eq(x**2-1, 0), Gt(x, 0)], (x,), dict=True) == [{x: 1}]
+    assert solve([Eq(x**3-6*x**2+11*x-6, 0), Eq(y**2, 1), Eq(z, 1)],
+            (x, y, z,), dict=True) == [{x: 1, y: -1, z: 1}, {x: 1, y: 1, z: 1},
+                                       {x: 2, y: -1, z: 1}, {x: 2, y: 1, z: 1},
+                                       {x: 3, y: -1, z: 1}, {x: 3, y: 1, z: 1}]
+    assert solve(Gt(x, 0), (x,), dict=True) == And(Lt(0, x), Lt(x, oo))
```
Comment (cbm755) [inline sympy/solvers/tests/test_solvers.py]: @oscarbenjamin this last test is likely what is throwing the UserWarning (?)

File: sympy/solvers/solvers.py [hunk 3]
```diff
@@ -983,6 +984,30 @@ def _sympified_list(w):
             f[i] = fi
 
         if fi.is_Relational:
+            if flags.get('dict', False):
+                solution = reduce_inequalities(f, symbols=symbols)
+                if isinstance(solution, Equality):
+                    return [{solution.lhs: solution.rhs}]
```
Comment (cbm755) [inline sympy/solvers/solvers.py]: this looks bad, were are enumerating over things.  We probably shouldn't return one of them...

Certainly no criticism intended to author of this.  In particular I see other suspicious looking returns up above too!  Line 973 and 975 for example...

**Comment by czgdp1807** [comment]
@cbm755 Let me know if a follow up PR is needed for this.

**Comment by cbm755** [comment]
I'm not sure, still learning here myself....    I guess it may get reverted for 1.5.

But I like your idea of having dict=True do its thing when it can!  But maybe its needs some revision...

**Comment by czgdp1807** [comment]
Refer, https://github.com/sympy/sympy/issues/11661#issuecomment-555362054

**Pull request merged**